Notebooks/notebook_temporary/utilities.py

```diff
@@ -63,3 +74,320 @@ def get_group_id(crop_group, crop_group_df, group_name='GROUP_1_NAME',
+
+    image = eopatch.data['BANDS'][time_frame_idx][..., [3, 2, 1]]
+
+    return np.clip(image * 3.5, 0, 1)
+
+
+def img_feature(eopatch, feature):
+    """Prepares data for selected timeless mask from given eopatch.
+
+    :param eopatch: Eopatch
+    :type eopatch: EOPatch
+    :param feature: Feature
+    :type feature: FeatureType
+    :return: Image data
+    :rtype: numpy.array
+    """
+    return eopatch[feature[0]][feature[1]].squeeze()
+
+
+def img_diff(eopatch, feature, reference_feature):
+    return eopatch.mask_timeless[reference_feature].squeeze() != eopatch.mask_timeless[feature].squeeze()
+
+
+def plot_roi(country, country_grid, eopatch_ids=None, title=None, size=20):
+    """Plot Region Of Interest.
+
+    :param country: Country data
+    :type country: GeoDataFrame
+    :param country_grid: Country split into bounding boxes, each representing
+        one EOPatch
+    :type country_grid: GeoDataFrame
+    :param eopatch_ids: 2D array of eopatch IDs, defaults to None
+    :type eopatch_ids: numpy.array, optional
+    :param title: Plot title, defaults to None
+    :type title: str, optional
+    :param size: plot size, defaults to 20
+    :type size: int, optional
+    """
+    roi_grid = None
+    
+    if eopatch_ids is not None:
+        roi_grid = country_grid.iloc[eopatch_ids.ravel()]
+
+    country_grid_x = country_grid['index_x'].max() + 1
+    country_grid_y = country_grid['index_y'].max() + 1
+
+    ratio = country_grid_y / country_grid_x
+
+    fig, ax = plt.subplots(figsize=(size, size * ratio))
+    country.plot(ax=ax, facecolor='#eef1f5', edgecolor='#666', alpha=1.0)
+    country_grid.plot(ax=ax, facecolor='#ffffff', edgecolor='#666', alpha=0.4)
+
+    if roi_grid is not None:
+        roi_grid.plot(ax=ax, facecolor='#f86a6a', edgecolor='#f86a6a', alpha=0.2)
+
+    for idx, row in country_grid.iterrows():
+        bbox = row.geometry
+        ax.text(
+            bbox.centroid.x,
+            bbox.centroid.y,
+            idx,
+            ha='center',
+            va='center',
+            fontdict={'fontsize': 8}
+        )
+
+    if title:
+        plot_title = title
+    else:
+        plot_title = f'{country.iloc[0]["name"]} ({len(country_grid)} eopatches)'
+
+    ax.set_title(plot_title, fontdict={'fontsize': 20})
+
+    plt.axis('off')
+    plt.tight_layout()
+
+
+def plot_features(path, features=None, size=20, max_cols=4):
+
+    eopatch = EOPatch.load(path, lazy_loading=True)
+
+    if features:
+        data_timeless_keys = features
+    else:
+        data_timeless_keys = list(eopatch.data_timeless.keys())
+
+    num_features = len(data_timeless_keys)
+
+    if not num_features:
+        print('Nothing to plot')
+        return
+
+    num_cols = min(max_cols, num_features)
+    num_rows = math.ceil(num_features / num_cols)
+
+    count = 0
+
+    fig = plt.figure(figsize=(20, 20 / num_cols * num_rows))
+    # fig.suptitle(f'eopatch_{idx}', fontsize=26)
```

Comment:
This should be handled through the arguments. Comments should be about explanations about the code to other developers.